### DISCUSSION TITLE
Go Pro service UUID FEA6 in 128-bit

### DISCUSSION THREAD

**Discussion by sultanmyrza** [gopro/OpenGoPro#41] [Q&A] [answered]
Originally I asked this question [here](https://github.com/gopro/OpenGoPro/issues/37) while @tcamise-gpsw answer was usefull to me I think my question was formulated not well.

> ### prehistory
>  I almost finished building ionic app that connects to GoPro via Bluetooth and WiFi then pull the photos to device which works for ios/android. Now after finishing MVP I decided to do some optimisation on Bluetooth search. P.S: I never worked with bluetooth things before it's my first bluetooth project. Sorry If my questions are dumb.

In ionic I use this plugin to search nearby bluetooth devices.
 
**This code snippet shows me all device nearby Bluetooth devices** 
```javascript
      const services: string[] = [];
      await BleClient.requestLEScan({ services }, (result)=>{ doSomething; });
```

then if we look at [GoPro Service UUIDS](https://gopro.github.io/OpenGoPro/ble#services-and-characteristics) table it says **Control & Query Service UUID** is `FEA6`

then I tried
```javascript
      const services: string[] = ['FEA6'];
      await BleClient.requestLEScan({ services });
```
#### **throws** `Invalid service UUID`

then I read again [GoPro Service UUIDS](https://gopro.github.io/OpenGoPro/ble#services-and-characteristics) table and noticed the note 😅
> Note: GP-XXXX is shorthand for GoPro's 128-bit UUIDs: b5f9xxxx-aa8d-11e3-9046-0002a5d5c51b

so FEA6 will be b5f9**fea6**-aa8d-11e3-9046-0002a5d5c51b ⁉️

then I tried 
```javascript
      const services: string[] = ['b5f9fea6-aa8d-11e3-9046-0002a5d5c51b'];
      await BleClient.requestLEScan({ services });
```
### no device found at all

then I scan again without providing services and show all Bluetooth devices then I connected to GoPro.
When I connected the output was something like
```
{
    "device": {
        "deviceId": "DE:CA:C7:C5:62:BA",
        "name": "GoPro 3543"
    },
    "localName": "GoPro 3543",
    "rssi": -59,
    "txPower": 127,
    "manufacturerData": {
        "754": {}
    },
    "serviceData": {
        "0000fea6-0000-1000-8000-00805f9b34fb": {}
    },
    "uuids": [
        "0000fea6-0000-1000-8000-00805f9b34fb"
    ],
    "rawAdvertisement": {}
}
```

### what is `0000fea6-0000-1000-8000-00805f9b34fb`? I don't know but now If I do
```javascript
      const services: string[] = ['0000fea6-0000-1000-8000-00805f9b34fb'];
      await BleClient.requestLEScan({ services });
```
### it shows only GoPro device

Here are my question

1. Is 0000fea6-0000-1000-8000-00805f9b34fb 128 bit version of FEA6?
2. I have only 1 GoPro will it find other nearby GoPros as well if I pass 0000fea6-0000-1000-8000-00805f9b34fb

**Comment by tcamise-gpsw** [comment, marked as answer]
@sultanmyrza I'm guessing that you already figured out the answers to these since you just merged your demo but, in any case:

1. I just found this [stack overflow answer](https://stackoverflow.com/questions/18699251/finding-out-android-bluetooth-le-gatt-profiles/20160164#20160164?newreg=2ff393507a45457aa4792e74da8a11e1) which says that 
   > all  BLE characteristic UUIDs are of the form: 0000XXXX-0000-1000-8000-00805f9b34fb

   I have never seen this before and do not see anywhere where this is actually defined by Bluetooth but it does appear to be true. Very strange...
2. If you are scanning by UUID, then yes it will find all devices (realistically only GoPro's) that use this UUID.
